Fix define command::executeResetSequence() to work with Oracle

CedricYii opened the pull request:
Introduce command::executeResetSequence() to handle OCI need for several
queries to reset a sequence.
Preserve backward compatibility with other databases.
Check if resetSequence() should be written as deprecated.

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes for travis
| Fixed issues  | #14289

CedricYii committed:
Fix define command::executeResetSequence() to work with Oracle

Introduce command::executeResetSequence() to handle OCI need for several
queries to reset a sequence.
Preserve backward compatibility with other databases.
Check if resetSequence() should be written as deprecated.

sergeymakinen reviewed (approved):
LGTM. `@since` should be added.

samdark commented:
@sergeymakinen should there be tests?

sergeymakinen commented:
@samdark Oh. I have noticed there's an existing test but forgot it should be altered. Good catch!

I will run tests as soon as they are ready.

sergeymakinen reviewed (changes requested):
Need updated tests for Oracle.

CedricYii committed:
Add command::executeResetSequence() to work with Oracle

Fix comments for default value.
Add @since2.0.16.
Remove db/oci/QueryBuilder::resetSequence().
Add tests.

CedricYii committed:
Add command::executeResetSequence() to work with Oracle

Fix comments for default value.
Add @since2.0.16.
Remove db/oci/QueryBuilder::resetSequence().
Add tests.

samdark commented:
Merged. Thank you!
